test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreNETCoreTest.cs

```diff
@@ -10001,6 +10001,59 @@ public void RestoreNetCore_WithMultipleFrameworksWithPlatformAndAssetTargetFallb
             }
         }
 
+        [PlatformFact(Platform.Windows)]
+        public async Task RestoreNetCore_MacCatalystUsingXamarinIOS_Warns()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                var maccatalyst = "net6.0-maccatalyst14.5";
+                var xamarinios = "Xamarin.iOS";
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+
+                packageX.AddFile($"lib/{xamarinios}/x.dll");
+
+                var projectA = SimpleTestProjectContext.CreateNETCore(
+                    "a",
+                    pathContext.SolutionRoot,
+                    maccatalyst
+                    );
+
+                var projectB = SimpleTestProjectContext.CreateNETCore(
+                    "b",
+                    pathContext.SolutionRoot,
+                    xamarinios
+                    );
+
+                projectA.AddPackageToAllFrameworks(packageX);
+                projectA.AddProjectToAllFrameworks(projectB);
+                solution.Projects.Add(projectA);
+                solution.Projects.Add(projectB);
+                solution.Create(pathContext.SolutionRoot);
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    pathContext.PackageSource,
+                    PackageSaveMode.Defaultv3,
+                    packageX);
+
+                // Act
+                var r = Util.RestoreSolution(pathContext);
+
+                // Assert
+                r.Success.Should().BeTrue();
+                r.AllOutput.Contains("WARNING: NU1901 - 'x' was resolved as a dependency of 'a', but the dependency is using 'Xamarin.iOS' while 'a' is using 'net6.0-maccatalyst14.5' as its TargetFramework");
+                r.AllOutput.Contains("WARNING: NU1901 - 'b' was resolved as a dependency of 'a', but the dependency is using 'Xamarin.iOS' while 'a' is using 'net6.0-maccatalyst14.5' as its TargetFramework");
```

Comment:
These aren't assertions, hence CI passed despite the NU codes were changed to `1703`

Reply:
omg ahahahahaha nice catch.